vllm/v1/engine/core.py

```diff
@@ -313,6 +325,15 @@ def step(self) -> tuple[dict[int, EngineCoreOutputs], bool]:
         if not self.scheduler.has_requests():
             return {}, False
         scheduler_output = self.scheduler.schedule()
+        ### nsys profiling
+        if self.profile_enabled:
+            self._profile_current_step = self._profile_current_step + 1
+            if self._profile_current_step in self.profiler_start_iters:
+                torch.cuda.cudart().cudaProfilerStart()
+
+            if self._profile_current_step in self.profiler_stop_iters:
+                torch.cuda.cudart().cudaProfilerStop()
+            self._profile_step_iteration_info(scheduler_output)
```

Comment:
Let's only log the step info within a certain profling range (e.g., 800 to 810).